mage_ai/frontend/utils/date.ts

```diff
@@ -78,6 +78,37 @@ export function datetimeInLocalTimezone(
   return datetime;
 }
 
+/**
+ * Given a UTC datetime string, find how much time has elapsed between then 
+ * and now. Return the elapsed time in the first matching format:
+ *   - >= 1 year: X year(s) ago
+ *   - >= 1 month: X month(s) ago
+ *   - >= 1 day: X day(s) ago
+ *   - >= 1 hr: X hr(s) Y min(s) ago
+ *   - < 1 hr: X min(s) ago
+ */
+export function utcStringToElapsedTime(datetime: string) {
+  const then = moment.utc(datetime);
+  const now = moment.utc();
+  const duration = moment.duration(now.diff(then));
+
+  let timeDisplay = '';
+  if (duration.years() >= 1) {
```

Comment:
@anniexcheng You can refactor this part to use the `pluralize` util method from the `mage_ai/frontend/utils/string.ts` file.